### ISSUE TITLE
Compile error occurs when an arg name is same as a package name.

### ISSUE THREAD

**Issue by yuzuy** [uber-go/mock#222]
**Actual behavior** A clear and concise description of what the bug is.

Compile error occurs when an arg name is same as a package name.

```go
import "github.com/yuzuy/xxx/pagination"

func (m *MockFooUseCase) Exec(ctx context.Context, pagination *pagination.Pagination) (*pagination.Page, error) {
	m.ctrl.T.Helper()
	ret := m.ctrl.Call(m, "Exec", ctx, pagination)
	ret0, _ := ret[0].(*pagination.Page) // pagination.Page is not a type
	ret1, _ := ret[1].(error)
	return ret0, ret1
}
```

**Expected behavior** A clear and concise description of what you expected to
happen.

Avoid a conflict between an arg name and a package name.

```go
import "github.com/yuzuy/xxx/pagination"

func (m *MockFooUseCase) Exec(ctx context.Context, paginationArg *pagination.Pagination) (*pagination.Page, error) {
	m.ctrl.T.Helper()
	ret := m.ctrl.Call(m, "Exec", ctx, paginationArg)
	ret0, _ := ret[0].(*pagination.Page)
	ret1, _ := ret[1].(error)
	return ret0, ret1
}
```

**To Reproduce** Steps to reproduce the behavior

-  Generate a mock of a method
    - that has an arg with the name that is same as the imported package name
    - that has a return value of the package type 

**Additional Information**

-   gomock mode (reflect or source): reflect
-   gomock version or git ref: v0.5.0
-   golang version: 1.23.1

**Triage Notes for the Maintainers**

**Comment by tompaz3** [comment]
I understand it's the issue for you (and probably for many others), but to be honest, I wouldn't personally introduce package name shadowing in the first place. Shadowing is unsafe on it's own and may lead to bugs eventually.

**Comment by rageagainsthepc** [comment]
@tompaz3 I agree, but unfortunately widely used packages like e.g. [docker](https://github.com/moby/moby/blob/master/client/interface.go#L74) seem to shadow their package names quite often. Right now this probably breaks a lot of projects using mockgen.

**Comment by tompaz3** [comment]
@rageagainsthepc Ah, I see. It's still doable, but not without the extra hussle (like replacing it with custom consumer-side interface).
Definitely agree that this is a breaking change, anyway.

**Comment by cavus700** [comment]
#219 Seems like this will be fixed in the next release

**Comment by yuzuy** [comment]
Thank you for fixing this issue!!!
#219 
https://github.com/uber-go/mock/releases/tag/v0.5.1
